src/app/provider.tsx

```diff
@@ -0,0 +1,18 @@
+// app/providers.tsx
+"use client";
+import { useState } from "react";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ReactQueryDevtools } from "@tanstack/react-query-devtools";
+
+export default function Providers({ children }: { children: React.ReactNode }) {
+  const [queryClient] = useState(() => new QueryClient());
```

Comment:
```suggestion
function makeQueryClient() {
  return new QueryClient({
    defaultOptions: {
      queries: {
        // SSR에서는 클라이언트에서 즉시 refetch하는 것을 피하기 위해
        // staleTime을 0보다 크게 설정하는 것이 좋다.
        staleTime: 60 * 1000,
      },
    },
  });
}

let browserQueryClient: QueryClient | undefined = undefined;

function getQueryClient() {
  if (typeof window === 'undefined') {
    // Server일 경우
    // 매번 새로운 queryClient를 만든다.
    return makeQueryClient();
  } else {
    // Browser일 경우
    // queryClient가 존재하지 않을 경우에만 새로운 queryClient를 만든다.
    // React가 새 Client를 만들게 하기 위해 중요하다.
    if (!browserQueryClient) browserQueryClient = makeQueryClient();
    return browserQueryClient;
  }
}
```


QueryClient는 싱글턴으로 유지하는 것이 더 나을 것 같아요. 클라이언트에서 매 요청마다 객체가 생성되는 것은 좀 별로니까요!

Reply:
[문서](https://tanstack.com/query/v4/docs/framework/react/guides/ssr#using-hydrate)를 참고하시면 nextJS의 서버 사이드 렌더링을 고려한 쿼리 로직을 hydration 개념과 합치실 수 있을 겁니다!